src/protections/NsfwProtection.ts

```diff
@@ -0,0 +1,85 @@
+import { Mjolnir } from "../Mjolnir";
+import * as nsfw from 'nsfwjs';
+import {LogLevel} from "matrix-bot-sdk";
+import { node } from '@tensorflow/tfjs-node';
+
+
+export class NsfwProtection extends Protection {
+    settings = {};
+    // @ts-ignore
+    private model: any;
+
+    constructor() {
+        super();
+    }
+
+    async initialize() {
+        this.model = await nsfw.load();
+    }
+
+    public get name(): string {
+        return 'NsfwProtection';
+    }
+
+    public get description(): string {
+        return "Scans all images sent into a protected room to determine if the image is " +
+            "NSFW. If it is, the image will automatically be redacted.";
+    }
+
+    public async handleEvent(mjolnir: Mjolnir, roomId: string, event: any): Promise<any> {
+        if (event['type'] === 'm.room.message') {
+            const content = event['content'] || {};
+            const msgtype = content['msgtype'] || 'm.text';
+            const formattedBody = content['formatted_body'] || '';
+            const isMedia = msgtype === 'm.image' || formattedBody.toLowerCase().includes('<img');
+
+            if (isMedia) {
+                const mxc = content["url"]
+                const image = await mjolnir.client.downloadContent(mxc)
+                const decodedImage = await node.decodeImage(image.data, 3);
+                const predictions = await this.model.classify(decodedImage)
+
+                for (const prediction of predictions) {
+                    if (prediction["className"] === "Porn") {
+                        if (prediction["probability"] > mjolnir.config.nsfwSensitivity) {
+                            await mjolnir.managementRoomOutput.logMessage(LogLevel.INFO, "NSFWProtection", `Redacting ${event["event_id"]} for inappropriate content.`);
+                            try {
+                                mjolnir.client.redactEvent(roomId, event["event_id"])
+                            } catch (err) {
+                                await mjolnir.managementRoomOutput.logMessage(LogLevel.ERROR, "NSFWProtection", `There was an error redacting ${event["event_id"]}: ${err}`);
+
+                            }
+                        }
+                    } else if (prediction["className"] === "Hentai") {
+                        if (prediction["probability"] > mjolnir.config.nsfwSensitivity) {
+                            await mjolnir.managementRoomOutput.logMessage(LogLevel.INFO, "NSFWProtection", `Redacting ${event["event_id"]} for inappropriate content.`);
+                            try {
+                                mjolnir.client.redactEvent(roomId, event["event_id"])
+                            } catch (err) {
+                                await mjolnir.managementRoomOutput.logMessage(LogLevel.ERROR, "NSFWProtection", `There was an error redacting ${event["event_id"]}: ${err}`);
+                            }
+                        }
+                    }
```

Comment:
we can reduce code duplication by combining the branches:
```suggestion
                    if (["Hentai", "Porn"].includes(prediction["className"])) {
                        if (prediction["probability"] > mjolnir.config.nsfwSensitivity) {
                            await mjolnir.managementRoomOutput.logMessage(LogLevel.INFO, "NSFWProtection", `Redacting ${event["event_id"]} for inappropriate content.`);
                            try {
                                mjolnir.client.redactEvent(roomId, event["event_id"])
                            } catch (err) {
                                await mjolnir.managementRoomOutput.logMessage(LogLevel.ERROR, "NSFWProtection", `There was an error redacting ${event["event_id"]}: ${err}`);

                            }
                        }
                    }
```